File: day1/assignment.cpp

// Q1: Program to calculate sum of first n natural numbers
 
#include <iostream>
using namespace std;
 
int main() {
    int n;
    cout << "Enter n: ";
    cin >> n;
    int sum = n * (n + 1) / 2;
    cout << "Sum of first " << n << " natural numbers = " << sum << endl;
    return 0;
}
 
 
// -------------------------------------------------------
 
 
// Q2: Program to print multiplication table of a given number
 
#include <iostream>
using namespace std;
 
int main() {
    int num;
    cout << "Enter a number: ";
    cin >> num;
    cout << "Multiplication table of " << num << ":" << endl;
    for (int i = 1; i <= 10; i++) {
        cout << num << " x " << i << " = " << num * i << endl;
    }
    return 0;
}
 
 
// -------------------------------------------------------
 
 
// Q3: Program to find factorial of a number
 
#include <iostream>
using namespace std;
 
int main() {
    int n;
    cout << "Enter a number: ";
    cin >> n;
    long long factorial = 1;
    for (int i = 1; i <= n; i++) {
        factorial *= i;
    }
    cout << "Factorial of " << n << " = " << factorial << endl;
    return 0;
}
 
 
// -------------------------------------------------------
 
 
// Q4: Program to count digits in a number
 
#include <iostream>
using namespace std;
 
int main() {
    long long num;
    cout << "Enter a number: ";
    cin >> num;
    int count = 0;
    if (num == 0) {
        count = 1;
    } else {
        if (num < 0) num = -num;
        while (num != 0) {
            num /= 10;
            count++;
        }
    }
    cout << "Number of digits = " << count << endl;
    return 0;
}
 

File: day1/assignment.py
# Q1: Program to calculate sum of first n natural numbers

n = int(input("Enter n: "))

# Using // for integer division to get a whole number
sum_val = n * (n + 1) // 2

print("Sum of first", n, "natural numbers =", sum_val)



# Q2: Program to print multiplication table of a given number

num = int(input("Enter a number: "))

print("Multiplication table of", num, ":")

# loop from 1 to 10
for i in range(1, 11):
    print(num, "x", i, "=", num * i)

 

# Q3: Program to find factorial of a number

n = int(input("Enter a number: "))

factorial = 1

# loop from 1 to n
for i in range(1, n + 1):
    factorial = factorial * i

print("Factorial of", n, "=", factorial)



# Q4: Program to count digits in a number

num = int(input("Enter a number: "))

count = 0

if num == 0:
    count = 1
else:
    # If the number is negative, make it positive
    if num < 0:
        num = -num
        
    while num != 0:
        num = num // 10  # Drops the last digit
        count = count + 1

print("Number of digits =", count)